lib/proxies/iot.spb.mqtt.js

```diff
@@ -0,0 +1,228 @@
+        "timestamp": new Date().getTime(),
+        "dataType": "String",
+        "value": client_data.swVersion || "SWDefaultv1.0"
+    }];  
+    
+    return metrics;
+};
+
+
+function IoTKitSparkplugBCloud(conf, broker) {
+
+    var me = this;
+    me.spbConf = conf.connector.mqtt;
+    me.client = broker;
+    me.type = 'mqtt';
+    me.topics = topic;
+    me.pubArgs = {
+        qos: 1,
+        retain: false
+    };
+    
+}
+
+IoTKitSparkplugBCloud.prototype.pullActuations = function (data, callback) {
+    callback(null);
+};
+
+/* For publishing sparkplugB standard Node Birth message
+* @devProf: Conatains all the device information and default component registered with 
+*   its component ids
+* Payload for device birth is by default created in function createNodeBirthMetrics
+*/
+
+IoTKitSparkplugBCloud.prototype.nodeBirth = function (devProf, callback) {
+    var me = this;
+    var topic = common.buildPath(me.topics.metric_topic, [me.spbConf.version,devProf.groupId,"NBIRTH",devProf.edgeNodeId, "" ]);
+    var client_data = {
+        "hwVersion" : null,
+        "swVersion" : null
+    }
+    var payload = {
+        "timestamp" : new Date().getTime(),
+        "metrics" : createNodeBirthMetrics(client_data),
+        "seq" : 0
+    };   
+   
+    return me.client.publish(topic, payload, me.pubArgs, function(err) {
+        if (err) {
+            return callback("fail");
+        } 
+        return callback({status : 0});
+    });
+   
+};
+
+/* For publishing sparkplugB standard device BIRTH messag
+* @devProf: Conatains all the device information and default component registered with 
+*   its component ids
+* Payload for device birth is in device profile componentMetric
+*/
+
+IoTKitSparkplugBCloud.prototype.deviceBirth = function (devProf, callback) {
+    var me = this;
+    var topic = common.buildPath(me.topics.metric_topic, [me.spbConf.version,devProf.groupId,"DBIRTH",devProf.edgeNodeId,devProf.deviceId]);
+    var payload = {
+        "timestamp" : new Date().getTime(),
+        "metrics" : devProf.componentMetric,
+        "seq" : 0
+    };   
+    return me.client.publish(topic, payload, me.pubArgs, function(err) {
+        if (err) {     
+            return callback("fail");
+        }
+        return callback({status : 0});
+    });
+};
+
+
+/* For publishing sparkplugB standard device DATA messag
+* @devProf: Contains all the device information and default component registered with 
+*   its component ids
+* @payloadMetric: Contains submitted data value in spB metric format to be sent to server
+*/
+IoTKitSparkplugBCloud.prototype.data = function (devProf,payloadMetric, callback) {
+    var me = this;   
+    var topic = common.buildPath(me.topics.metric_topic, [me.spbConf.version,devProf.groupId,"DDATA",devProf.edgeNodeId,devProf.deviceId]);  
+    var payload = {
+        "timestamp" : new Date().getTime(),
+        "metrics" : payloadMetric,
+        "seq" : incSeqNum()
```

Comment:
What is the purpose of 'seq' in the SpB message? It looks to start at 0 with the birth message and subsequent messages it gets incremented until it hits 255 where it returns to 0? Is it really that simple or does it need more advanced handling?